pkg/core/dao/dao.go

```diff
@@ -69,6 +72,16 @@ func (dao *Simple) GetPrivate() *Simple {
 	st := storage.NewPrivateMemCachedStore(dao.Store)
 	d := newSimple(st, dao.Version.StateRootInHeader, dao.Version.P2PSigExtensions)
 	d.Version = dao.Version
+	if dao.keyBuf != nil { // This one is private.
```

Comment:
I got it, only private DAOs do have `dao.keyBuf != nil`, right?

Reply:
They're derived from shared one, so each instance has a cache of their own. But then we add another layer on top of `cache` for executions and that's where I'd like to avoid any additional allocations for sure. Other than that we don't have a lot of `GetPrivate` users, I think they're fine.